k2/csrc/array.h

```diff
@@ -49,6 +55,40 @@ class Array1 {
                                        byte_offset_);
   }
 
+  // Return a reference to this viewed as `Any` type (for when we
+  // want a generic array without type informatiopn)
+  // Reverse this with Specialize().
+  Array1<Any> &Generic() {
+    return *reinterpret_cast<Array1<Any>*>(this);
+  }
+  Array1<Any> &Generic() const {
```

Comment:
I am afraid this won't compile if you instantiate it.

It is not possible to bind a `const lvalue reference` to a `non-const lvalue reference`.

It also shows that a large part of the change is not covered by unittests.